### DISCUSSION TITLE
The "^" is magic and needs an example

### DISCUSSION THREAD

**Discussion by AddictArts** [vercel/turborepo#2084] [Help] [answered]
I am reading and re-reading multiple times "The `^` symbol explicitly declares that the task has a dependency on a task in a workspace it depends on."

Without an example or better explanation I am unclear exactly the behavior. The declaration  `"dependsOn": ["^build"]` has no other build pipelines named build to depend on. Also in other workspaces the build commands in scripts can't declare dependencies they don't have `turbo.json's` to map dependencies.

The issue an poor clarity is that `dependsOn` is used to mention other pipelines, but with `^` it seems to be changing the semantics and imply some sort of workspace dependency pipeline that is implicit. It might be better to have a different key?

I think what it is saying is that if a workspace package depends on another local workspace package it will build that first? At least an example would make it explicitly clear. I'd prefer not to make assumptions on what will happen.

Note: To make matters worse, at least  for me, recent youtube "How many apps can you buld in one Turborepo?" the author never adds this magic `^` or the `dependsOn` in the build pipeline. I thin the reason is none of the apps depend on each other. Turbo is fairly explicit and clear, easy to reason about expect the magic `^`.

**Comment by mehulkar** [comment, marked as answer]
> I think what it is saying is that if a workspace package depends on another local workspace package it will build that first?

That's correct! We've updated the docs recently around this, hopefully it's helpful: https://turbo.build/repo/docs/core-concepts/monorepos/running-tasks#in-a-different-workspace. Could you open an issue (or if you're feeling adventurous, a PR) if this still doesn't feel clear enough?
